tests/async.rs

```diff
@@ -93,24 +93,27 @@ where
 
     fn test<C, F>(self, c: C)
     where
-        F: IntoFuture<Item = (), Error = mysql_async::errors::Error>,
+        F: IntoFuture<Item = (), Error = mysql_async::error::Error>,
         C: FnOnce(mysql_async::Conn) -> F,
     {
+        let mut runtime = current_thread::Runtime::new().unwrap();
+
         let listener = net::TcpListener::bind("127.0.0.1:0").unwrap();
         let port = listener.local_addr().unwrap().port();
+
         let jh = thread::spawn(move || {
             let (s, _) = listener.accept().unwrap();
             MysqlIntermediary::run_on_tcp(self, s)
         });
 
-        let mut core = Core::new().unwrap();
-        let handle = core.handle();
-        let db =
-            core.run(mysql_async::Conn::new(
-                &format!("mysql://127.0.0.1:{}", port),
-                &handle,
-            )).unwrap();
-        core.run(c(db).into_future()).unwrap();
+        let pool = mysql_async::Pool::new(format!("mysql://127.0.0.1:{}", port));
+
+        let future = pool.get_conn().and_then(|conn| c(conn));
```

Comment:
Come to think of it, I think I'd just stick this directly into the `block_on` call below.

Reply:
Ah, that really cleans things up. This way doesn't require remembering to drop/disconnect the pool.